Adjust some nullpo checks

MrKeiKun opened the pull request:
<!-- Before you continue, please change "base: stable" to "base: master" and
     enable the setting "[√] Allow edits from maintainers." when creating your
     pull request if you have not already enabled it. -->

<!-- Note: Lines with this <!-- syntax are comments and will not be visible in
     your pull request. You can safely ignore or remove them. -->

### Pull Request Prelude

<!-- Thank you for working on improving Hercules! -->
<!-- Please complete these steps and check the following boxes by putting an `x`
     inside the [brackets] before filing your Pull Request. -->

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

- #2973
Adjust some nullpo check on status_calc_regen_pc and status_calc_regen_rate_pc
<!-- Describe the changes that this pull request makes. -->

**Issues addressed:**
#2973
#2847 (probably?)  <!-- Write here the issue number, if any. -->


<!-- You can safely ignore the links below:  -->

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

hemagx commented on src/map/status.c:
```diff
@@ -2608,8 +2608,11 @@ static void status_calc_regen_pc(struct map_session_data *sd, struct status_data
-	nullpo_retv(regen->sitting);
-	nullpo_retv(regen->skill);
+
+	if (regen->skill == NULL)
```
The original design here is that `regen->skill`, `regen->sitting`  should never be null, but now you're defaulting to make it `sd` is there a rationale behind it? the commit messages doesn't give a rationale at all or explain how it fixes this issue, this seem more like a band-aid workaround than an actual fix for the issue mentioned.

MrKeiKun replied:
you are kinda right here. its a bandaid solution. maybe i should solve the real problem.

hemagx commented on src/map/status.c:
```diff
@@ -2683,38 +2686,42 @@ static void status_calc_regen_pc(struct map_session_data *sd, struct status_data
-	struct regen_data_sub *skill_regen = regen->skill;
-	skill_regen->hp = 0;
-	skill_regen->sp = 0;
+	if (regen->skill != NULL) {
```
It cannot be null still with your check this is redundant.

MrKeiKun replied:
same here.

MrKeiKun committed:
Fixed null regen->skill and regen->sitting crash on player login

In pc_authok, pc_setpos is called to warp the player to their last saved
position before status_calc_pc(SCO_FIRST) runs in pc_reg_received. Since
pc_setpos triggers status_calc_regen, and SCO_FIRST is the only place that
initializes sd->regen.skill and sd->regen.sitting, those pointers were NULL
on the first regen calculation, causing a nullpo crash. Initializing both
pointers in pc_authok before the pc_setpos call fixes the root cause.

Signed-off-by: Lorenzo Buitizon <the.keikun@gmail.com>
